features/dd-sdk-android-session-replay-compose/src/test/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/AbstractSemanticsNodeMapperTest.kt

```diff
@@ -84,6 +84,19 @@ internal open class AbstractCompositionGroupMapperTest {
         testedMapper = StubAbstractSemanticsNodeMapper(mockSemanticsUtils, mockColorStringFormatter)
     }
 
+    @Test
+    fun `M return correct Id W resolveId`(@IntForgery fakeIndex: Int) {
+        // Given
+        val semanticsNode = mock<SemanticsNode>()
+        whenever(semanticsNode.id) doReturn fakeSemanticsId
+
+        // When
+        val result = testedMapper.stubResolveId(semanticsNode, fakeIndex)
+        val expected = (fakeSemanticsId.toLong() shl 32) + fakeIndex
```

Comment:
maybe we can also do the verification the other way: here we are copying what it is already in the production code, but if we think that the requirement is "to have 64 bits composite number: most significant 32 bits is A, least significant 32 bits is B", then we can extract `A` and `B` from `result` and compare them independently.

Not a big deal though.

Reply:
indeed, that can protect better the code than copying the production one, updated